api/datasciencecluster/v1/datasciencecluster_conversion.go

```diff
@@ -17,27 +17,111 @@ limitations under the License.
 package v1
 
 import (
+	"strings"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/api/common"
 	dscv2 "github.com/opendatahub-io/opendatahub-operator/v2/api/datasciencecluster/v2"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
+var componentNameMapping = map[string]string{
+	"DataSciencePipelines": "AIPipelines",
+}
+
+// convertConditionType converts a condition type by replacing v1 component name with v2 or vice versa
+// using the componentNameMapping.
+func convertConditionType(condType, from, to string) string {
+	return strings.ReplaceAll(condType, from, to)
+}
+
+// convertConditionsV1ToV2 converts condition types from v1 to v2 by replacing
+// any occurrence of component names using componentNameMapping.
+func convertConditionsV1ToV2(v1Conditions []common.Condition) []common.Condition {
+	if v1Conditions == nil {
+		return nil
+	}
+	v2Conditions := make([]common.Condition, len(v1Conditions))
+	for i, cond := range v1Conditions {
+		v2Conditions[i] = cond
+		condType := cond.Type
+		// Replace all v1 component names with v2 component names
+		for v1Name, v2Name := range componentNameMapping {
+			condType = convertConditionType(condType, v1Name, v2Name)
+		}
+		v2Conditions[i].Type = condType
+	}
+	return v2Conditions
+}
+
+// convertConditionsV2ToV1 converts condition types from v2 to v1 by replacing
+// any occurrence of component names using componentNameMapping (reversed).
+func convertConditionsV2ToV1(v2Conditions []common.Condition) []common.Condition {
+	if v2Conditions == nil {
+		return nil
+	}
+	v1Conditions := make([]common.Condition, len(v2Conditions))
+	for i, cond := range v2Conditions {
+		v1Conditions[i] = cond
+		condType := cond.Type
+		// Replace all v2 component names with v1 component names (reverse mapping)
+		for v1Name, v2Name := range componentNameMapping {
+			condType = convertConditionType(condType, v2Name, v1Name)
+		}
+		v1Conditions[i].Type = condType
+	}
+	return v1Conditions
+}
+
```

Comment:
Those functions are quite the same, we could write once passing params on name to convert. So, there will be a `convertConditions(conditions []common.Condition, from, to string)`.